src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolModel.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal class DocumentSymbolModel
```

Comment:
```suggestion
    internal sealed class DocumentSymbolModel
```

Reply:
nit: it concerns me that this seems to be mutable.  It means we have a mutable object that is being worked on by the BG workers in the controller.  It would be much cleaner if we have a split from immutable data model, and the mutable UI model.  Then all work on the BG is inherently safe as it literally cannot be mutating shared state.

Reply:
Let's talk about this offline :)